webrender_traits/src/types.rs

```diff
@@ -743,3 +743,55 @@ pub enum WebGLShaderParameter {
     Bool(bool),
     Invalid,
 }
+
+#[repr(C)]
+pub struct ExternalImageKey(pub u32);
+
+pub enum ExternalImage<'i> {
+    TextureHandle {
+        handle: u32,
+    },
+    MemoryBuff {
+        buff: &'i [u8],
+    }
+}
+
+#[repr(C)]
+pub enum ExternalImageType {
+    TEXTURE_HANDLE,
+    MEM_OR_SHMEM,
+}
+
+#[repr(C)]
+pub struct ExternalImageStruct {
```

Comment:
How about:

```rust
struct RgbImage {
    size: Size2D<u32>,
    stride: u32,
    format: ImageFormat,
    buffer: *const u8,
}

struct GlHandle {
     handle: u32,
}

// YuvImage, DxgiHandle, etc...

enum ExternalImage {
    RgbMemory(RgbImage),
    YuvMemory(YuvImage),
    Gl(GlHandle),
    Dxgi(DxgiHandle),
    // etc..
}
```

Using enums this way is a lot more idiomatic in rust,  and I think that it'll make it easier to work with internally